src/types/signer-options.ts

```diff
@@ -45,25 +45,37 @@ const SessionOptionsSchema = z.object({
   sessionPermissionExpirationInMilliseconds: z.number().positive().optional()
 });
 
-export const SignerOptionsSchema = z.object({
+export const SignerInitOptionsSchema = z.object({
```

Comment:
Why are you creating a duplicate of the options?

Why are host and sessionOptions provided in both SignerInitOptionsSchema and SignerOptionsSchema?

If this is necessary, can at least parts of it be provided in a separate PR with related tests?